mlrun/runtimes/nuclio/application/application.py

```diff
@@ -527,21 +561,22 @@ def _build_application_image(
             with_mlrun=with_mlrun,
         )
 
-    def _ensure_reverse_proxy_configurations(self):
-        if self.spec.build.functionSourceCode or self.status.container_image:
+    @classmethod
+    def _ensure_reverse_proxy_configurations(cls, function):
```

Comment:
I think this method should be a static method rather than a class method because it does not change any class state or use any class data. 
Also, `get_filename_and_handler` should also be a static method for the same reason.